src/components/lib/cos_kernel_api.c

```diff
@@ -486,7 +570,7 @@ __page_bump_alloc(struct cos_compinfo *ci, size_t sz)
 		if (!umem) return 0;
 
 		/* Actually map in the memory. */
-		if (call_cap_op(meta->mi.pgtbl_cap, CAPTBL_OP_MEMACTIVATE, umem, ci->pgtbl_cap, heap_cursor, 0)) {
+		if (call_cap_op(meta->mi.pgtbl_cap, CAPTBL_OP_MEMACTIVATE, umem, ci->pgtbl_cap, heap_cursor, PAGE_ORDER)) {
```

Comment:
all of this code around page sizes should be generic.  One set of functions that are parameterized by the order/size of pages to be operated on.  We cannot accept a large level of code replication.